lib/msf/core/db_manager/event.rb

```diff
@@ -12,8 +12,8 @@ def report_event(opts = {})
     return if not wspace # Temp fix?
     uname  = opts.delete(:username)
 
-    if ! opts[:host].kind_of? ::Mdm::Host and opts[:host]
-      opts[:host] = report_host(:workspace => wspace, :host => opts[:host])
+    if !opts[:host].nil? && !opts[:host].kind_of?(::Mdm::Host)
+      opts[:host] = find_or_create_host(workspace: wspace, host: opts[:host])
```

Comment:
This is going to call `Msf::DBManager::Host.find_or_create_host` and not the proxy version. We basically have two methods that do the same thing, with different logic. This is something that is a problem across all of our models, and I don't think we should do anything about this here, but it just doesn't feel right duplicating these actions on the server. Shouldn't it always be on the server or always on the client?

Reply:
I don't disagree regarding the data proxy and DBManager logic. If DbManager code is executing and all calls within DBManager do not call out to the data proxy, then the calls would remain local to the client and server.